solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -407,25 +519,41 @@ def process_forecast_observations(forecast_observations, filters, data,
                                     for k, v in counts.items())
                 preproc_results = (datamodel.PreprocessingResult(
                     name=VALIDATION_RESULT_TOTAL_STRING,
-                    count=len(data[fxobs.data_object]) - len(obs_ser)), )
+                    count=int(len(data[fxobs.data_object]) - len(obs_ser))), )
                 validated_observations[fxobs.data_object] = (
                     obs_ser, val_results, preproc_results)
 
         obs_ser, val_results, preproc_results = (
             validated_observations[fxobs.data_object])
 
-        # resample and align observations to forecast, create
-        # ProcessedForecastObservation
+        # Apply fill to forecasts
+        freq = fxobs.forecast.interval_length
         fx_ser = data[fxobs.forecast]
+        fx_ser, count = apply_fill(fx_ser, missing_forecast, start, end, freq)
+        if _get_dtype(fx_ser) == np.object:
```

Comment:
what's the purpose of the if statement? wondering if there's something that should be addressed somewhere else.

Reply:
Yeah. This one took me a while to figure out and only caught with the full report test. 

If the forecast is empty, the data that is returned is an empty series/dataframe with no assigned `dtype` which by default is just `numpy.object`.  [See here](https://pandas.pydata.org/docs/whatsnew/v1.0.0.html#default-dtype-of-empty-pandas-series). 

I was thinking maybe we can force the `dtype` upon return from the api such as in `io.api.get_forecast_values` but that would mean we would have to know the `dtype` say from the type of forecast, so I didn't see way around it right now so instead used the `dtype` of the `Observation` so at least process proceeds. Perhaps it would be better to raise an error?

Reply:
That's surprising since there's a type assignment here: https://github.com/SolarArbiter/solarforecastarbiter-core/blob/88fe63e2c8a92dbb9913bdc20875dfcbd4e514fd/solarforecastarbiter/io/utils.py#L152

I think at this level of the code we can assume float too.

Reply:
I also don't understand why `fx_ser` would be empty unless `missing_forecast == 'drop'` in which case you can assign the dtype in `apply_fill`

Reply:
If `fx_ser` comes into the function as empty with `object` dtype, I would think this would already give us problems

Reply:
That's probably it, but I can't recall anymore. I'll certainly change it to be `float`. `numpy.nan` has to be a float anyways.

Reply:
I reran all the tests but can't reproduce the error. I also don't have any check after `drop` is applied. But just for safe measure I'll add an `astype(float)` conversion after application.